src/server/server_family.cc

```diff
@@ -1867,11 +1878,8 @@ void ServerFamily::Info(CmdArgList args, ConnectionContext* cntx) {
       append("replication_full_sync_buffer_bytes", repl_mem.full_sync_buf_bytes_);
     }
 
-    {
-      lock_guard lk{save_mu_};
-      if (save_controller_) {
-        append("save_buffer_bytes", save_controller_->GetSaveBuffersSize());
-      }
+    if (has_save_controller) {
+      append("save_buffer_bytes", save_controller_->GetSaveBuffersSize());
```

Comment:
you need to lock save_mu_ before accessing save_controller_

Reply:
omg :scream: :scream: good catch!